pkg/controller/accountclaim/reuse.go

```diff
@@ -265,6 +267,83 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccountS3(reqLogger logr.Logger, awsCl
 	return nil
 }
 
+func (r *ReconcileAccountClaim) cleanUpAwsRoute53(reqLogger logr.Logger, awsClient awsclient.Client, awsNotifications chan string, awsErrors chan string) error {
+
+	var nextZoneMarker *string
+
+	// Paginate through hosted zones
+	for {
+		// Get list of hosted zones by page
+		hostedZonesOutput, err := awsClient.ListHostedZones(&route53.ListHostedZonesInput{Marker: nextZoneMarker})
+		if err != nil {
+			listError := "Failed to list Hosted Zones"
+			awsErrors <- listError
+			return err
+		}
+
+		for _, zone := range hostedZonesOutput.HostedZones {
+
+			// List and delete all Record Sets for the current zone
+			var nextRecordName *string
+			// Pagination again!!!!!
+			for {
+				recordSet, listRecordsError := awsClient.ListResourceRecordSets(&route53.ListResourceRecordSetsInput{HostedZoneId: zone.Id, StartRecordName: nextRecordName})
+				if listRecordsError != nil {
+					recordSetListError := fmt.Sprintf("Failed to list Record sets for hosted zone %s", *zone.Name)
+					awsErrors <- recordSetListError
+					return listRecordsError
+				}
+
+				changeBatch := &route53.ChangeBatch{}
+				for _, record := range recordSet.ResourceRecordSets {
+					// Build ChangeBatch
+					// https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#ChangeBatch
+					//https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#Change
+					if *record.Type == "TXT" {
```

Comment:
Considering the DeleteHostedZone call will fail if there are records different from NS/SOA in a hosted zone, shouldn't we clean up all records that are not NS/SOA here, instead of just the TXT ones?

Reply:
That way we avoid failing reuse in case customers add CNAMEs/ALIAS for example.

Reply:
:+1:

Reply:
:+1: We should check and delete all records in the hosted zone (except NS/SOA of course).